identity-iota/src/credential/resolved_credential.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::credential_validator::CredentialValidator;
+use super::CredentialValidationOptions;
+use crate::document::ResolvedIotaDocument;
+use crate::Result;
+use identity_credential::credential::Credential;
+use serde::Serialize;
+
+/// A verifiable credential whose associated DID documents have been resolved from the Tangle.
+#[non_exhaustive]
+pub struct ResolvedCredential<T> {
+  pub credential: Credential<T>,
+  pub issuer: ResolvedIotaDocument,
+}
```

Comment:
Whether or not we keep `ResolvedCredential` and `ResolvedPresentation` depends on the intended high-level API. Right now I believe we could replace these with methods on the `Resolver` like `resolve_credential_issuer`, `resolve_presentation_issuers` and `resolve_presentation_holder` without losing ergonomics and it would work around ownership rules.

Separating `resolve_presentation_issuers` and `resolve_presentation_holder` may be pertinent for ZKP since, as was pointed out on Discord, the holder may not necessarily be identifiable from the presentation or its proof.

However, if we want to have an "all-in-one, resolve and validate this presentation" function we likely need a way for developers to access those resolved DID Documents of the issuers and holders.

Reply:
If `ResolvedCredential` and `ResolvedPresentation` are not used in this PR I think we can remove them, perhaps, and revisit the problem with the `Resolver` PR?

Reply:
Good idea. I will remove them.